subsystem: net/lib/wifi_credentials: Switch to PSA ITS (#13838)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Added more default LTE metrics, such as band, operator, RSRP, and kilobytes sent and received.
  * Updated the default metric names to follow the standard |NCS| variable name convention.

> * :ref:`lib_wifi_credentials` library:

Reviewer: Could move to the "Libraries for networking" section.

>   * Updated PSA backend to use PSA Internal Trusted Storage to store Wi-Fi credentials instead of Protected Storage.
>     This was changed because PSA Internal Trusted Storage is better suited to store secrets.
>     When switching storage the credentials will need to be migrated manually, or the existing credentials will be lost.
>     Removed :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_UID_OFFSET`.
>     Use :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_OFFSET` instead.

Reviewer: ```suggestion
  * Updated PSA backend to use PSA Internal Trusted Storage (ITS) to store Wi-Fi credentials instead of protected storage.
    The change is because PSA ITS is better suited to store secrets.
    When switching storage the credentials need to be migrated manually, or the existing credentials are lost.
    Removed ``CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_UID_OFFSET``.
    Use the :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_OFFSET` Kconfig option instead.
```
